presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1031,6 +1032,371 @@ public void testVirtualBucketing()
         }
     }
 
+    @Test
+    public void testMaterializedViewOptimization()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE orders_partitioned WITH (partitioned_by = ARRAY['ds']) AS " +
+                    "SELECT orderkey, orderpriority, '2020-01-01' as ds FROM orders WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey > 1000");
+
+            assertUpdate("CREATE MATERIALIZED VIEW test_orders_mv WITH (partitioned_by = ARRAY['ds']) AS SELECT orderkey, orderpriority, ds FROM orders_partitioned");
+            assertTrue(getQueryRunner().tableExists(getSession(), "test_orders_mv"));
+            assertUpdate("INSERT INTO test_orders_mv(orderkey, orderpriority, ds) select orderkey, orderpriority, ds from orders_partitioned where ds='2020-01-01'", 255);
+
+            String mvQuery = "SELECT orderkey from test_orders_mv where orderkey <  10000";
+            String baseQuery = "SELECT orderkey from orders_partitioned where orderkey <  10000";
+            System.out.println("Explain plan is: \n" + getExplainPlan(mvQuery, LOGICAL));
```

Comment:
Let's clean the tests by \:
- disable them with the REFRESH PR issue number as the reason
- make sure we spell out `mv` into `view` or `materializedView`
- Remove all System.out.println
- (Major) can we assert the plan structure just like all other tests do?

Reply:
yeah I wanted to disable the tests just before we merge.